### DISCUSSION TITLE
Problemas al ejecutar comando pip install -r requirements.txt

### DISCUSSION THREAD

**Discussion by DanielUTN** [frlp-utn-ingsoft/catedra#85] [Q&A] [answered]
Buenas tardes Profesor,
     Le consulto porque me aprecio el siguiente error al ejecutar el comando pip install -r requirements.txt
![al ejecutar requirements](https://github.com/frlp-utn-ingsoft/catedra/assets/108563375/3a203dc0-3805-4e6b-ada3-0c0d8113df3a)

He encontrado algunas opciones para arreglarlo en stackoverflow, pero antes igual le consulto alguna via de solucion por si acaso.
Actualice el pip a la ultima version. 
Debo de usar un entorno virtual ?

Saludos

**Comment by Salani1003** [comment]
Proba con el comando 
```pip uninstall --force-all-dependencies sqlparse```

y despues proba devuelta 

```pip install -r requirements.txt```

Por ultimo, yo no se si los profes lo recomiendan, pero siempre que trabajo con python uso entornos virtuales. 
Espero haberte ayudado

**Reply by RodrigoJacznik** [reply, marked as answer]
Gracias por la respuesta. Si, lo mejor es trabajar con entornos virtuales, en la clase no lo nombramos para no complicar las cosas.

Para crear un entorno virtual:
`python -m venv .venv`

Para activarlo:
`source .venv/bin/activate`

Para desactivarlo
`deactivate`

Lo que hay que tener en cuenta es que siempre que se quiera correr la app o correr los test hay que estar en el entorno virtual activado.
Las dependencias también tienen que estar instaladas en el entorno virtual.

**Reply by DanielUTN** [reply]
Gracias por las respuestas!. Cree un entorno virtual.
Saludos
